src/JustEat.StatsD/V2/UdpTransportV2.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Net.Sockets;
+using JustEat.StatsD.EndpointLookups;
+
+#if !NET451
+using System.Runtime.InteropServices;
+#endif
+
+namespace JustEat.StatsD.V2
+{
+    public sealed class UdpTransportV2 : IStatsDTransportV2
+    {
+        private readonly IPEndPointSource _endpointSource;
+
+        public UdpTransportV2(IPEndPointSource endPointSource)
+        {
+            _endpointSource = endPointSource ?? throw new ArgumentNullException(nameof(endPointSource));
+        }
+
+        public void Send(ArraySegment<byte> metric)
```

Comment:
As per previous comments, we should be able to make the existing types implement both ways.